include/moveit_grasps/two_finger_grasp_generator.h

```diff
@@ -0,0 +1,240 @@
+   * \param alignment_rotation - extra rotatation needed to align grasp pose as you move around the cuboid
+   * \param num_grasps - the number of grasps to generate around the corner
+   * \param grasp_poses_tcp - list of grasp poses generated
+   * \return the number of poses generated
+   */
+  std::size_t addFaceGraspsHelper(const Eigen::Isometry3d& pose, double rotation_angles[3],
+                                  const Eigen::Vector3d& translation, const Eigen::Vector3d& delta,
+                                  double alignment_rotation, std::size_t num_grasps,
+                                  EigenSTL::vector_Isometry3d& grasp_poses_tcp);
+
+  /**
+   * \brief helper function for adding grasps along the edges of the cuboid
+   * \param pose - pose of the object to grasp
+   * \param rotation_angles - rotation angles to go from cuboid pose to standard grasping pose
+   * \param delta - distance to move away from cuboid at each step
+   * \param translation - translation to go from cuboid centroid to grasping location
+   * \param alignment_rotation - extra rotatation needed to align grasp pose as you move around the cuboid
+   * \param num_grasps - the number of grasps to generate around the corner
+   * \param grasp_poses_tcp - list of grasp poses generated
+   * \return the number of poses generated
+   */
+  std::size_t addEdgeGraspsHelper(const Eigen::Isometry3d& cuboid_pose, double rotation_angles[3],
+                                  const Eigen::Vector3d& translation, const Eigen::Vector3d& delta,
+                                  double alignment_rotation, std::size_t num_grasps,
+                                  EigenSTL::vector_Isometry3d& grasp_poses_tcp, double corner_rotation);
+
+  /**
+   * \brief helper function for determining if the grasp will intersect the cuboid
+   * \param cuboid_pose - centroid of object to grasp in world frame
+   * \param depth - size of cuboid along x axis
+   * \param width - size of cuboid along y axis
+   * \param height - size of cuboid along z axis
+   * \param grasp_pose_tcp - pose of grasp
+   * \param grasp_data - data describing end effector
+   * \return true if the grasp intersects the cuboid
+   */
+  bool graspIntersectionHelper(const Eigen::Isometry3d& cuboid_pose, double depth, double width, double height,
+                               const Eigen::Isometry3d& grasp_pose_tcp, const TwoFingerGraspDataPtr& grasp_data);
+
+  /**
+   * \brief helper function to test intersection of a line with a plane
+   * \param t - parametric distance along grasp line
+   * \param u1, v1, u2, v2 - (u,v) coordinates of the line
+   * \param a, b - length and width of the plane area in which to test for intersection
+   * \param u, v - location of intersection
+   * \return true if the line intersects the plane
+   */
+  bool intersectionHelper(double t, double u1, double v1, double u2, double v2, double a, double b, double& u,
```

Comment:
I've avoided touching this part of the code because of exactly this reason. There is a ton of simple math with useless variable names that could be made much more user friendly. For now I just want to refactor out the finger and suction grippers so that they are separate.